common/k8s/src/middleware/metadata.rs

```diff
@@ -112,6 +124,12 @@ impl Middleware for K8sMetadata {
                 let obj_ref =
                     ObjectRef::new(&parse_result.pod_name).within(&parse_result.pod_namespace);
                 if let Some(pod) = self.store.get(&obj_ref) {
+                    let meta_object =
+                        extract_image_name_and_tag(parse_result.container_name, pod.as_ref());
+                    if line.set_meta(json!(meta_object)).is_err() {
+                        return Status::Skip;
```

Comment:
just skip?

Reply:
following the pattern of how the rest of the fields are set.

Reply:
how about logging the error? if meta got lost then the line that gets sent is invalid.

Reply:
I would normally agree but the error that is returned from this is actually nothing. It would be just as helpful to not have a log for it. 

However - in the spirit of this i am going to add 

> This looks great, can we get an integration test for it? Or maybe just add the check to one of the existing kubernetes ones?

added

Reply:
@dkhokhlov added some more logic around your suggestion.